pkg/util/winutil/service.go

```diff
@@ -0,0 +1,241 @@
+		if err != nil {
+			return fmt.Errorf("could not retrieve service status: %v", err)
+		}
+	}
+	return nil
+}
+
+func doStopService(serviceName string) error {
+	return ControlService(serviceName, svc.Stop, svc.Stopped, windows.SERVICE_STOP|windows.SERVICE_QUERY_STATUS, defaultServiceCommandTimeout)
+}
+
+func StopService(serviceName string) error {
+
+	deps, err := ListDependentServices(serviceName, windows.SERVICE_ACTIVE)
+	if err != nil {
+		return fmt.Errorf("could not list dependent services: %v", err)
+	}
+
+	for _, dep := range deps {
+		err = doStopService(dep.serviceName)
+		if err != nil {
+			return fmt.Errorf("could not stop service %s: %v", dep.serviceName, err)
+		}
+	}
+	return doStopService(serviceName)
+}
+
+func RestartService(serviceName string) error {
+	var err error
+	if err = StopService(serviceName); err == nil {
+		err = StartService(serviceName)
+	}
+	return err
+}
+
+// when Go has their version, replace ours with the upstream
+// https://github.com/golang/go/issues/56766
+func ListDependentServices(serviceName string, state enumServiceState) ([]EnumServiceStatus, error) {
+	m, err := OpenSCManager(windows.SC_MANAGER_CONNECT)
+	if err != nil {
+		return nil, err
+	}
+	defer m.Disconnect()
+
+	s, err := OpenService(m, serviceName, windows.SERVICE_ENUMERATE_DEPENDENTS)
+	if err != nil {
+		return nil, fmt.Errorf("could not open service: %v", err)
+	}
+	defer s.Close()
+
+	deps, err := enumDependentServices(s.Handle, state)
+	if err != nil {
+		return nil, fmt.Errorf("could not enumerate dependent services: %v", err)
+	}
+	return deps, nil
+}
+
+// ServiceStatus reports information pertaining to enumerated services
+// only exported so binary.Read works
+type ServiceStatus struct {
+	DwServiceType             uint32
+	DwCurrentState            uint32
+	DwControlsAccepted        uint32
+	DwWin32ExitCode           uint32
+	DwServiceSpecificExitCode uint32
+	DwCheckPoint              uint32
+	DwWaitHint                uint32
+}
+
+// EnumServiceStatus complete enumerated service information
+// only exported so binary.Read works
+type EnumServiceStatus struct {
+	serviceName string
+	displayName string
+	status      ServiceStatus
+}
+
+type internalEnumServiceStatus struct {
+	ServiceName uint64 // offset from beginning of buffer
+	DisplayName uint64 // offset from beginning of buffer.
+	Status      ServiceStatus
+	Padding     uint32 // structure is qword aligned.
+
+}
+
+func enumDependentServices(h windows.Handle, state enumServiceState) (services []EnumServiceStatus, err error) {
+	services = make([]EnumServiceStatus, 0)
+	var bufsz uint32
+	var count uint32
+	_, _, err = procEnumDependentServices.Call(uintptr(h),
+		uintptr(state),
+		uintptr(0),
+		uintptr(0), // current buffer size is zero
+		uintptr(unsafe.Pointer(&bufsz)),
+		uintptr(unsafe.Pointer(&count)))
+
+	// success with a 0 buffer means no dependent services
+	if err == error(windows.ERROR_SUCCESS) {
+		err = nil
+		return
+	} else {
```

Comment:
There no need for else here after return above, right

Reply:
Yep, good catch, will update.